Tests/RegressionTestMessageHandler.cs

```diff
@@ -0,0 +1,240 @@
+        }
+
+        private void UpdateRegressionStatisticsInSourceFile(BacktestResultPacket result)
+        {
+            var algorithmSource = Directory.EnumerateFiles("../../../Algorithm.CSharp", $"{_job.AlgorithmId}.cs", SearchOption.AllDirectories).Single();
+            var file = File.ReadAllLines(algorithmSource).ToList().GetEnumerator();
+            var lines = new List<string>();
+            while (file.MoveNext())
+            {
+                var line = file.Current;
+                if (line == null)
+                {
+                    continue;
+                }
+
+                if (line.Contains("Dictionary<string, string> ExpectedStatistics => new Dictionary<string, string>")
+                    || line.Contains("Dictionary<string, string> ExpectedStatistics => new()"))
+                {
+                    if (!result.Results.Statistics.Any())
+                    {
+                        lines.Add(line);
+                        continue;
+                    }
+
+                    lines.Add(line);
+                    lines.Add("        {");
+
+                    foreach (var pair in result.Results.Statistics)
+                    {
+                        lines.Add($"            {{\"{pair.Key}\", \"{pair.Value}\"}},");
+                    }
+
+                    // remove trailing comma
+                    var lastLine = lines[lines.Count - 1];
+                    lines[lines.Count - 1] = lastLine.Substring(0, lastLine.Length - 1);
+
+                    // now we skip existing expected statistics in file
+                    while (file.MoveNext())
+                    {
+                        line = file.Current;
+                        if (line != null && line.Contains("};"))
+                        {
+                            lines.Add(line);
+                            break;
+                        }
+                    }
+                }
+                else if (line.Contains($"long DataPoints =>"))
+                {
+                    lines.Add(GetDataPointLine(line, _algorithmManager?.DataPoints.ToString()));
+                }
+                else if (line.Contains($"int AlgorithmHistoryDataPoints =>"))
+                {
+                    lines.Add(GetDataPointLine(line, _algorithmManager?.AlgorithmHistoryDataPoints.ToString()));
+                }
+                else
+                {
+                    lines.Add(line);
+                }
+            }
+
+            file.DisposeSafely();
+            File.WriteAllLines(algorithmSource, lines);
+        }
+
+        private string GetDataPointLine(string currentLine, string count)
+        {
+            var dataParts = currentLine.Split(" ");
+            dataParts[^1] = count + ";";
+            return string.Join(" ", dataParts);
+        }
+
+        /// <summary>
+        /// Dispose of any resources
+        /// </summary>
+        public void Dispose()
+        {
+        }
+    }
```

Comment:
Ideally we should inherit form existing `MessageHandler` and only implement the statistics updates, with no duplication. Also the updating can now be removed from the `MessageHandler`?